### ISSUE TITLE
Training whit prodigy annotations 

### ISSUE THREAD

**Issue by alejandrojcastaneira** [explosion/spaCy#3428]
Good afternoon. As always impressed whit all your releases, and big fan of your work.

I have been doing an annotation project also using Prodigy and Spacy, exporting the anotations to spacy whit the built in recipe gold-to-spacy and training whit the spacy=2.0.18, waiting for an stable release of the new 2.1!. 

Yesterday, didn't hesitate into trying the 2.1 and try to train whit  my custom ner using a blank("en") model. The training scrip similar to the one is in the docs, just adding:

    for text,annotations in pre_train_data:

        doc = nlp.make_doc(text)

        gold = GoldParse(doc, entities=annotations)

        sample = [doc, gold]

        train_data.append(sample)

before starting the training, because I was exporting whit --biluo, in spacy=2.0.18 works excellent.
now I'm getting:

"File "gold.pyx", line 565, in spacy.gold.GoldParse.__init__
IndexError: list index out of range"

then I tried to train from character offsets removing the --biluo tag and the described code, the training starts but show this before finishing the 1th iteration:

"[E024] Could not find an optimal move to supervise the parser. Usually, this means the GoldParse was not correct. For example, are all labels added to the model?"

I understand the tokenization between models could be different, I also tried it to check the unicode built of python: 1114111 just in case, how could I train whit the Prodigy annotations?

Best regards

**Comment by honnibal** [comment]
Hmm. Do any of your entity annotations start or end with whitespace? That could be the problem with the error you're seeing during training. If so we can definitely improve either the error message or the behaviour there.

Could you figure out which sentence it fails on? That would make it much easier to figure out what's going wrong.

Another option is to convert into spaCy's json format, so you can use the `spacy train` command. You should be able to do `python -m spacy convert -l en -t json -c jsonl /input-data.jsonl /output_dir`. It should work on the file you exported from Prodigy.

**Comment by alejandrojcastaneira** [comment]
I debugged today, in the texts that it stops, there was one entitity " Javascript" which started whit a white space, so that might be the problem. 

I corrected the annotation then I tried to train from the command line and keep  showing the [E024] error, then I try to train using the corrected annotation from offsets and the regular scrip and it works! will be good if we could handle those cases  (also I can make and script to check these cases or better prepossessing my input), also I might have other entities which include \n inside, not sure if that matters.
